// login/tailwind.config.js
/** @type {import('tailwindcss').Config} */
export default {
  darkMode: "class",
  content: [
    "./index.html",
    "./src/**/*.{js,ts,jsx,tsx}",
  ],
  theme: {
    extend: {
      colors: {
        "tertiary-container": "#a7b1d9",
        "on-secondary-container": "#005669",
        "secondary-fixed-dim": "#3cd7ff",
        "primary-fixed-dim": "#38debb",
        "tertiary-fixed-dim": "#bbc5ee",
        "primary-container": "#00c9a7",
        "on-primary": "#00382d",
        "inverse-primary": "#006b58",
        "surface": "#0b1325",
        "primary": "#44e5c2",
        "on-tertiary": "#252f50",
        "error": "#ffb4ab",
        "on-tertiary-fixed": "#0f1a39",
        "outline": "#85948e",
        "on-primary-fixed": "#002019",
        "on-error": "#690005",
        "background": "#0b1325",
        "on-primary-container": "#004e40",
        "secondary-container": "#00d2fd",
        "on-tertiary-fixed-variant": "#3c4567",
        "on-secondary-fixed-variant": "#004e5f",
        "on-background": "#dbe2fb",
        "inverse-on-surface": "#283043",
        "on-surface-variant": "#bacac3",
        "on-surface": "#dbe2fb",
        "secondary": "#a2e7ff",
        "surface-bright": "#31394c",
        "on-error-container": "#ffdad6",
        "tertiary-fixed": "#dce1ff",
        "tertiary": "#c3ccf6",
        "on-secondary-fixed": "#001f27",
        "inverse-surface": "#dbe2fb",
        "surface-container-highest": "#2d3448",
        "secondary-fixed": "#b4ebff",
        "surface-container-low": "#131b2d",
        "surface-container": "#171f32",
        "surface-container-lowest": "#060e1f",
        "on-secondary": "#003642",
        "surface-tint": "#38debb",
        "primary-fixed": "#5ffbd6",
        "surface-dim": "#0b1325",
        "surface-container-high": "#222a3d",
        "on-tertiary-container": "#3a4365",
        "surface-variant": "#2d3448",
        "error-container": "#93000a",
        "on-primary-fixed-variant": "#005142",
        "outline-variant": "#3c4a45"
      },
      fontFamily: {
        "headline": ["Inter", "sans-serif"],
        "body": ["Inter", "sans-serif"],
        "label": ["Inter", "sans-serif"]
      },
      borderRadius: {"DEFAULT": "0.125rem", "lg": "0.25rem", "xl": "0.5rem", "full": "0.75rem"},
    },
  },
  plugins: [],
}
